homeassistant/components/shelly/binary_sensor.py

```diff
@@ -268,10 +268,8 @@ class RestBinarySensor(ShellyRestAttributeEntity, BinarySensorEntity):
     entity_description: RestBinarySensorDescription
 
     @property
-    def is_on(self) -> bool | None:
+    def is_on(self) -> bool:
         """Return true if REST sensor state is on."""
-        if self.attribute_value is None:
-            return None
```

Comment:
Is this not possible?

Reply:
I think this code was added before we had typing for `aioshelly`, this is the code that create the sensor (there is only 1 `REST` sensor for now):
https://github.com/home-assistant/core/blob/d9aabaea04c13e2582de085e1c9a61cd2282a56a/homeassistant/components/shelly/binary_sensor.py#L138-L146

For this to happen we need to have a situation that all the keys are valid but the value of the key is `None`, it can only happen due to a bug in the device firmware and in this case I think it would be better to let it raise an error, but I don't think this could ever happen.